src/entities/encoders/composableStable.ts

```diff
@@ -50,17 +51,16 @@ export class ComposableStableEncoder {
         amounts: AddLiquidityAmounts & { maxAmountsInWithoutBpt: bigint[] },
     ): Address {
         switch (kind) {
-            case AddLiquidityKind.Init:
-                throw new Error(
-                    'For this kind use initPool instead of addLiquidity',
-                );
             case AddLiquidityKind.Unbalanced:
                 return ComposableStableEncoder.addLiquidityUnbalanced(
                     amounts.maxAmountsInWithoutBpt,
                     amounts.minimumBpt,
                 );
             case AddLiquidityKind.SingleToken: {
-                if (amounts.tokenInIndex === undefined) throw Error('No Index');
+                // just a sanity check as this is already checked in InputValidator
+                if (amounts.tokenInIndex === undefined) {
+                    throw addLiquiditySingleTokenShouldHaveTokenInIndexError;
+                }
```

Comment:
Do we still need it then? 😉

Reply:
There are a few ways to avoid it, but they are not necessarily better than the current solution 😅 
1. Add a bang operator - I wouldn't go that way, because in case we ever reach that error, it would fail without a proper message
2. Refactor the getAmountsCall to be specific for each RemoveLiquidityKind - which could increase the complexity given RemoveLiquidityCall is already a union type handling different pool types 😬

Do you have a preferred solution? Or even a 3rd option? 🙃

Reply:
Ahh sorry, I meant if its already checked in InputValidator why do we have to check it again.

Reply:
Since this is non critical, I'll push in order to get things going. We can always refactor in a later PR. 🙃

Reply:
Makes sense, thanks! Seems good to just stick as is.